geode-core/src/main/java/org/apache/geode/internal/cache/GemFireCacheImpl.java

```diff
@@ -1086,6 +1088,32 @@ public static GemFireCacheImpl getForPdx(String reason) {
     clientMetadataService = clientMetadataServiceFactory.apply(this);
   }
 
+  public void lockDiskStore(String diskStoreName) {
+    CountDownLatch countDownLatch = diskStoreLatches.get(diskStoreName);
+    if (countDownLatch == null) {
+      countDownLatch = diskStoreLatches.putIfAbsent(diskStoreName, new CountDownLatch(1));
+      if (countDownLatch != null) {
+        try {
+          countDownLatch.await();
+        } catch (InterruptedException e) {
+          throw new InternalGemFireError(e);
+        }
+      }
+    } else {
+      try {
+        countDownLatch.await();
+      } catch (InterruptedException e) {
+        throw new InternalGemFireError(e);
+      }
+    }
+  }
+
+  public void unlockDiskStore(String diskStoreName) {
+    if (diskStoreLatches.get(diskStoreName) != null) {
+      diskStoreLatches.get(diskStoreName).countDown();
+    }
```

Comment:
I think there is a small possibility of a race condition causing an NPE here if the latch associated with `diskStoreName` is removed after the first `get()` call. Would it be possible to change this method to use only one `get()` call?

Reply:
I second @DonalEvans's comments here, could we use `computeIfPresent` or something similar?.
As a side note, if somebody invokes `unlockDiskStore` before calling `lockDiskStore` for exactly the same `diskName` first (aka the entry is `null`), shouldn't we throw an exception directly as the caller is using the API in a wrong way?.

Reply:
Agree. There could be an NPE.

Reply:
@jujoramos if the entry is `null`, we can't tell whether it is because the disk store is removed, or somebody invokes `unlockDiskStore` before `lockDiskStore`. We don't want to throw an exception, when the entry is `null` because the disk store is removed.